pypeit/core/save.py

```diff
@@ -10,133 +10,49 @@
-                tdict = dict(RA=0., DEC=0.,  # J2000
-                             objid=specobj.objid, slitid=specobj.slitid, det=det, scidx=specobj.scidx,  # specobj IDs
-                             FWHM=0.,  # Spatial resolution in arcsec
-                             R=0.,     # Spectral resolution (FWHM) in lambda/Dlambda
-                             xslit=specobj.xslit, nslit=sl+1)  # Slit position and number
-                meta.add_row(tdict)
-    # Remove dummy row and write
-    meta = meta[1:]
-    hdf['meta'] = meta
-
-    # Make a Header from fitsdict
-    hdict = {}
-    for key in fitsdict.keys():
-        hdict[key] = fitsdict[key][slf._specobjs[detref][0][0].scidx]  # Hopefully this is the right index
-    d = linetools.utils.jsonify(hdict)
-    hdf['header'] = json.dumps(d)
-
-    # Loop on extraction methods
-    for ex_method in ['boxcar', 'optimal']:
-        # Check for extraction type
-        if not hasattr(slf._specobjs[detref][0][0], ex_method):
-            continue
-        method_grp = hdf.create_group(ex_method)
-
-        # Data arrays are always MaskedArray
-        dtypes = []
-        for key in getattr(slf._specobjs[detref][0][0], ex_method).keys():
-            dtype = 'float64' if key == 'wave' else 'float32'
-            dtypes.append((str(key), dtype, (totpix)))
-        dtypes.append((str('obj_trace'), 'float32', (totpix)))
-        data = np.ma.empty((1,), dtype=dtypes)
-        # Setup in hdf5
-        spec_set = hdf[str(ex_method)].create_dataset('spec', data=data, chunks=True,
-                                                      maxshape=(None,), compression='gzip')
-        spec_set.resize((nspec,))
-        # Fill (and make meta)
-        count = 0
-        for kk in range(settings.spect['mosaic']['ndet']):
-            det = kk+1
-            if slf._specobjs[det - 1] is None:
-                continue
-            # Loop on slits
-            for sl in range(len(slf._specobjs[det - 1])):
-                nspec += len(slf._specobjs[det - 1][sl])
-                # Loop on spectra
-                for specobj in slf._specobjs[det-1][sl]:
-                    # Check meta
-                    assert meta['objid'][count] == specobj.objid
-                    # Trace
-                    data['obj_trace'][0][:len(specobj.trace)] = specobj.trace
-                    # Rest
-                    sdict = getattr(specobj, ex_method)
-                    for key in sdict.keys():
-                        npix = len(sdict[key])
-                        try:
-                            data[key][0][:npix] = sdict[key].value
-                        except AttributeError:
-                            data[key][0][:npix] = sdict[key]
-                    # Write
-                    spec_set[count] = data
-                    count += 1
-    #
-    hdf.close()
-
-    # Dump into a linetools.spectra.xspectrum1d.XSpectrum1D
-'''
+    Routine to save PypeIt 1d and 2d outputs
+    Args:
+        sci_dict: dict
```

Comment:
I don't like this terse documentation because it does not clarify image sizes and dimensionality. We should discuss at the next telecon. While it makes no difference in this case, it does make a a difference when dimensionality or size of an image need to be clarified. I still don't feel that I have a consistent set of examples of how this documentation should look which is why I'm not exactly writing it in that format.

Reply:
In this case, it is the identical info. 
Anyways, we aren't going to change Google's style:

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Reply:
The rest of the docs in PypeIt are in shambles so I'm not sure why you are being so strict about this now. We clearly need to completely overhaul docs but I disagree with delaying this PR to do so.

Reply:
Given I spent several days working on docs at the top level,
I don't appreciate that work being referred to as `shambles`.
Try again.